internal/install/recipes.go

```diff
@@ -14,15 +14,31 @@ type recipeFetcher interface {
 }
 
 type recipe struct {
-	Name              string                 `yaml:"name"`
-	Description       string                 `yaml:"description"`
-	Repository        string                 `yaml:"repository"`
-	Platform          string                 `yaml:"platform"`
-	Arch              string                 `yaml:"arch"`
-	TargetEnvironment string                 `yaml:"target_environment"`
-	ProcessMatch      []string               `yaml:"process_match"`
-	MELTMatch         meltMatch              `yaml:"melt_match"`
-	Install           map[string]interface{} `yaml:"install"`
+	InputVars    []variableConfig       `yaml:"inputVars"`
+	Install      map[string]interface{} `yaml:"install"`
+	Keywords     []string               `yaml:"keywords"`
+	MELTMatch    meltMatch              `yaml:"meltMatch"`
+	MetaData     metaData               `yaml:"metadata"`
+	ProcessMatch []string               `yaml:"processMatch"`
+	Variant      variant                `yaml:"variant"`
+}
+
+type metaData struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+	Repository  string `yaml:"repository"`
+}
+
+type variant struct {
+	OS                []string `yaml:"os"`
+	Arch              []string `yaml:"arch"`
+	TargetEnvironment []string `yaml:"targetEnvironment"`
+}
+
+type variableConfig struct {
+	Name    string `yaml:"name"`
+	Prompt  string `yaml:"prompt"`
+	Default string `yaml:"default"`
 }
```

Comment:
@zlesnr This is still a _little_ off from what the spec is. 

Should look more like this I think:
```
type recipe struct {
	MetaData     metaData               `yaml:"metadata"`
	InputVars    []variableConfig       `yaml:"inputVars"`
	Install      map[string]interface{} `yaml:"install"`
}

type metaData struct {
	Name          string 	`yaml:"name"`
	Description   string 	`yaml:"description"`
	Repository    string 	`yaml:"repository"`
	Keywords      []string  `yaml:"keywords"`
	MELTMatch     meltMatch	`yaml:"meltMatch"`
	ProcessMatch  []string  `yaml:"processMatch"`
	Variant       variant   `yaml:"variant"`
}

```

For ref: [amazonLinux2.yml recipe](https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/infra-agent/amazonlinux2.yml)